src/GovUk.Education.ExploreEducationStatistics.Admin/Services/FileUploadsValidatorService.cs

```diff
@@ -195,13 +114,95 @@ private bool IsFileExisting(Guid releaseId, ReleaseFileTypes type, string name)
+            
+            if (FileContainsSpacesOrSpecialChars(dataFileName))
+            {
+                return ValidationActionResult(FilenameCannotContainSpacesOrSpecialCharacters);
+            }
+            
+            if (metaFileName.IndexOf(" ", Ordinal) > -1 || 
+                metaFileName.IndexOfAny(Path.GetInvalidFileNameChars()) > -1)
+            {
+                return ValidationActionResult(MetaFilenameCannotContainSpacesOrSpecialCharacters);
+            }
+
+            if (FileContainsSpacesOrSpecialChars(metaFileName))
+            {
+                return ValidationActionResult(FilenameCannotContainSpacesOrSpecialCharacters);
+            }
+            
+            if (!metaFileName.ToLower().Contains(".meta."))
+            {
+                return ValidationActionResult(MetaFileIsIncorrectlyNamed);
+            }
+
+            if (IsFileExisting(releaseId, ReleaseFileTypes.Data, dataFileName))
+            {
+                return ValidationActionResult(CannotOverwriteDataFile);
+            }
+
+            if (IsFileExisting(releaseId, ReleaseFileTypes.Metadata, metaFileName))
+            {
+                return ValidationActionResult(CannotOverwriteMetadataFile);
+            }
+
+            return true;
+        }
+
+        private async Task<Either<ActionResult, bool>> ValidateDataFileSizes(long dataFileLength, long metaFileLength)
+        {
+            if (dataFileLength == 0)
+            {
+                return ValidationActionResult(DataFileCannotBeEmpty);
+            }
+
+            if (metaFileLength == 0)
+            {
+                return ValidationActionResult(MetadataFileCannotBeEmpty);
+            }
+
+            return true;
+        }
         
-        public static readonly Dictionary<ReleaseFileTypes, IEnumerable<Regex>> AllowedMimeTypesByFileType = 
-            new Dictionary<ReleaseFileTypes, IEnumerable<Regex>>
+        private async Task<Either<ActionResult, bool>> ValidateDataFileTypes(Guid releaseId, IFormFile dataFile, IFormFile metaFile)
+        {
+            var dataFilePath = AdminReleasePath(releaseId, ReleaseFileTypes.Data, dataFile.FileName.ToLower());
+            var metadataFilePath = AdminReleasePath(releaseId, ReleaseFileTypes.Data, metaFile.FileName.ToLower());
+
+            if (!IsCsvFile(dataFilePath, dataFile))
             {
-                { ReleaseFileTypes.Ancillary, AllowedAncillaryFileTypes },
-                { ReleaseFileTypes.Chart, AllowedChartFileTypes },
-                { ReleaseFileTypes.Data, AllowedCsvMimeTypes }
-            };
+                return ValidationActionResult(DataFileMustBeCsvFile);
+            }
+
+            if (!IsCsvFile(metadataFilePath, metaFile))
+            {
+                return ValidationActionResult(MetaFileMustBeCsvFile);
+            }
+
+            return true;
+        }
+        
+        private async Task<Either<ActionResult, bool>> ValidateSubjectName(Guid releaseId, string name)
```

Comment:
If the Either's right is a `Unit` you don't need to return a `true`.

Reply:
done